### PR TITLE
Teach ComparerEngine to not choke on native binaries

### LINKED ISSUE #47: Comparing LibGit2Sharp blows with "Format of the executable (.exe) or library (.dll) is invalid."
To repro just execute

http://localhost:2321/compare/LibGit2Sharp/0.20.1...0.20.2

causes Mono.Cecil to blow with mentioned ex.

### PR DESCRIPTION
**Pull request by nulltoken**
Fix #47 

Some additional proposals have been added. Feel free to tell me if you'd prefer me to drop them. I'll be happy to comply :wink:

### CODE REVIEW

**Comment by nulltoken** [comment]
Still a WIP. I'll work on the fix *per se* tomorrow

**Comment by nulltoken** [comment]
@andreasohlund @SimonCropp No longer a WIP, ready for review.

File: APIComparer.Core/ComparerEngine.cs [hunk 1]
```diff
@@ -22,30 +24,49 @@ public Diff CreateDiff(string leftAssembly, string rightAssembly)
...
+        IEnumerable<TypeDefinition> ReadTypes(string leftAssembly, bool readSymbols)
         {
-            var readerParams = new ReaderParameters
+            return ReadTypes(new[] { leftAssembly }, readSymbols);
+        }
+
+        private bool IsManagedAssembly(string assemblyPath)
+        {
+            // From https://github.com/scriptcs/scriptcs/pull/250/files by @aaronpowell
+            // License: Apache 2.0 (https://github.com/scriptcs/scriptcs/blob/dev/LICENSE.md)
```
Comment (nulltoken) [inline APIComparer.Core/ComparerEngine.cs]: /cc @aaronpowell I've ~~shamelessly stolen~~ been greatly inspired by your work in https://github.com/scriptcs/scriptcs/pull/250. Are you cool with this?

**Comment by nulltoken** [comment]
A different approach would be to actually move this `IsManagedAssembly()` call in `NuGetDownloader.DownloadAndExtractVersion()`. Would you prefer this one?

File: APIComparer.Backend.Tests/APIComparer.Backend.Tests.csproj [hunk 2]
```diff
@@ -69,6 +85,9 @@
   <ItemGroup>
     <Folder Include="Properties\" />
   </ItemGroup>
+  <ItemGroup>
+    <Service Include="{82A7F48D-3B50-4B1E-B82E-3ADA8210C358}" />
```
Comment (andreasohlund) [inline APIComparer.Backend.Tests/APIComparer.Backend.Tests.csproj]: Is this some VS auto add junk?

**Comment by andreasohlund** [comment]
Since we also support , use the .exe, comparing local files as well I think the current solution is the best one

**Comment by nulltoken** [comment]
@andreasohlund Duh. I didn't notice your comment earlier and just pushed a fixup to showcase the alternative approach. Still preferring the first one?

File: APIComparer.Backend.Tests/APIComparer.Backend.Tests.csproj (hunk 2 above)
Comment (nulltoken) [inline APIComparer.Backend.Tests/APIComparer.Backend.Tests.csproj]: This is some marker that VisualStudio automatically adds to identify test projects to only load the Test Explorer when that makes sense (see http://stackoverflow.com/a/22917902/335418).

**Comment by nulltoken** [comment]
I'm going to rollback the fixup as indeed this wouldn't help wouldn't we rely on the `NuGetDownloader` types. BTW, why do we have two of them?

**Comment by andreasohlund** [comment]
Yea, I prefer the first version

Regarding the duplication, I think they will diverge soon so we kept them separate for now

Copy paste reuse ftw!;)

File: APIComparer.Backend.Tests/APIComparer.Backend.Tests.csproj (hunk 2 above)
Comment (nulltoken) [inline APIComparer.Backend.Tests/APIComparer.Backend.Tests.csproj]: I've eventually dropped it from this PR. This was adding far too much noise to the diff.

**Pull request merged**